### PR TITLE
[DSHARE-1072] prevent zero amount revert due to low token precision input

### PR DESCRIPTION
**Pull request by joshualyguessennd**
This [PR](https://dinari.atlassian.net/jira/software/projects/DSHARE/boards/2?assignee=712020%3Aa20e54a0-5e4c-4203-9d5b-a3911c614d59&selectedIssue=DSHARE-1072) adds bounds to test amounts:
- Minimum > 1 to ensure non-zero amounts after price conversion
- Maximum < uint256.max/2 to prevent potential overflows

These bounds help tests mimic real-world token transfers while preventing edge cases that would never occur in production.

### CODE REVIEW

**Review by ykim — APPROVED** [review]

**Review by jaketimothy — APPROVED** [review]
This is better, but many of the failures will still happen with this change. Not sure what the right number is without testing a range of oracle price inputs

File: test/UsdPlusRedeemer.t.sol [hunk 1]
```diff
@@ -269,7 +269,7 @@ contract UsdPlusRedeemerTest is Test {
     }
 
     function test_permitRequestRedeem(uint256 amount) public {
-        vm.assume(amount > 0 && amount < type(uint256).max / 2);
```
Comment (ching) [inline test/UsdPlusRedeemer.t.sol:272]: This only updates the test cases - where is the actual changes?

Comment (joshualyguessennd) [reply test/UsdPlusRedeemer.t.sol:272]: this is only updating test cases because the issue isn't a bug in the contract code - it's about making our tests more realistic. The zero amount revert is correct contract behavior, we're just improving our tests to not trigger this expected revert with unrealistic amounts

Comment (ching) [reply test/UsdPlusRedeemer.t.sol:272]: The PR title implies it is fixing actual code instead of tests. Would be more clear if the PR title include the note about fixing this issue raised by test cases.
